llvm/lib/Transforms/Scalar/Scalarizer.cpp

```diff
@@ -699,6 +728,18 @@ bool ScalarizerVisitor::splitCall(CallInst &CI) {
   if (isVectorIntrinsicWithOverloadTypeAtArg(ID, -1))
     Tys.push_back(VS->SplitTy);
 
+  if (AreAllVectors) {
+    Type *PrevType = CallType->getContainedType(0);
+    Type *CallType = CI.getType();
```

Comment:
nit: Is there a reason for this to be re-defined? I don't see any side-effects in `getVectorSplit/isStructAllVectors`

Reply:
Nice catch, this is left over from a previous version where`CallType`  was `CI.getType()` or `CI.getType()->getContainedType(0);` and for this case I needed the struct to iterate over. Later change I inited VS as `getVectorSplit(CallType);` or `VS(getVectorSplit(CallType->getContainedType(0));` and thus didn't need to do this, but forgot to remove it.